server/pulp/server/controllers/content.py

```diff
@@ -128,113 +137,392 @@ def refresh_content_source(content_source_id=None):
     step.process_lifecycle()
 
 
-@celery.task(base=Task)
-def _download_one(catalog_entry):
+def queue_download_deferred():
     """
-    Download the content unit specified by the catalog entry.
-
-    :param catalog_entry: The catalog entry to download content for.
-    :type  catalog_entry: pulp.server.db.model.LazyCatalogEntry
+    Queue a task to download all content units with entries in the DeferredDownload
+    collection.
     """
-    _logger.info(_('Downloading {url} via the Pulp streamer').format(url=catalog_entry.url))
-    try:
-        if os.path.exists(catalog_entry.path):
-            # Already downloaded
-            return
-
-        _download_catalog_entry(catalog_entry)
-    except PluginNotFound:
-        msg = _('Download of {url} failed - plugin not found')
-        _logger.info(msg.format(url=catalog_entry.url))
-    except DoesNotExist:
-        msg = _('Download of {url} failed - unit not found')
-        _logger.info(msg.format(url=catalog_entry.url))
-    except ModelNotFound, e:
-        msg = _('Download {url}, failed - {reason}')
-        _logger.info(msg.format(url=catalog_entry.url, reason=str(e)))
+    tags = [pulp_tags.action_tag(pulp_tags.ACTION_DEFERRED_DOWNLOADS_TYPE)]
```

Comment:
I feel like the tags on both these new tasks are not the best so I welcome other suggestions. Maybe this one should carry the 'schedule' resource tag?